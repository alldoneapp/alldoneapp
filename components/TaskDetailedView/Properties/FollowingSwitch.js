import React, { Component } from 'react'
import { StyleSheet, Text, View, Animated } from 'react-native'
import styles, { colors } from '../../styles/global'
import { TouchableWithoutFeedback } from 'react-native-gesture-handler'

export default class FollowingSwitch extends Component {
    constructor(props) {
        super(props)

        this.state = {
            yes: false,
            marginLeft: new Animated.Value(1),
            backgroundColorAnim: new Animated.Value(0),
            textColorAnim: new Animated.Value(0),
        }
    }

    render() {
        let backgroundColor = this.state.backgroundColorAnim.interpolate({
            inputRange: [0, 1],
            outputRange: [colors.Gray300, colors.Primary300],
        })
        let textColor = this.state.textColorAnim.interpolate({
            inputRange: [0, 1],
            outputRange: [colors.Text02, colors.Primary300],
        })
        return (
            <View style={localStyles.container}>
                <View style={localStyles.textContainer}>
                    <Animated.Text style={[this.state.yes ? styles.subtitle1 : styles.body1, { color: textColor }]}>
                        {this.state.yes ? 'Yes' : 'No'}
                    </Animated.Text>
                </View>
                <TouchableWithoutFeedback onPress={this.onPress}>
                    <Animated.View
                        style={[
                            localStyles.switchContainer,
                            {
                                paddingLeft: this.state.marginLeft,
                                backgroundColor: backgroundColor,
                                borderColor: backgroundColor,
                            },
                        ]}
                    >
                        <View style={localStyles.lever}></View>
                    </Animated.View>
                </TouchableWithoutFeedback>
            </View>
        )
    }

    onPress = () => {
        if (this.state.yes) {
            Animated.parallel(
                [
                    Animated.timing(this.state.marginLeft, {
                        toValue: 1,
                        duration: 125,
                    }),
                    Animated.timing(this.state.backgroundColorAnim, {
                        toValue: 0,
                        duration: 125,
                    }),
                    Animated.timing(this.state.textColorAnim, {
                        toValue: 0,
                        duration: 125,
                    }),
                ],
                { stopTogether: false }
            ).start()
        } else {
            Animated.parallel(
                [
                    Animated.timing(this.state.marginLeft, {
                        toValue: 19,
                        duration: 125,
                    }),
                    Animated.timing(this.state.backgroundColorAnim, {
                        toValue: 1,
                        duration: 125,
                    }),
                    Animated.timing(this.state.textColorAnim, {
                        toValue: 1,
                        duration: 125,
                    }),
                ],
                { stopTogether: false }
            ).start()
        }

        this.setState({ yes: !this.state.yes })
    }
}

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'row',
        alignItems: 'center',
    },
    switchContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        width: 42,
        height: 24,
        paddingRight: 1,
        borderWidth: 1,
        borderRadius: 21,
        //borderColor: colors.Primary300,
        //backgroundColor: colors.Primary300,
    },
    lever: {
        width: 20,
        height: 20,
        backgroundColor: 'white',
        borderRadius: 100,
    },
    textContainer: {
        marginRight: 12,
    },
})
