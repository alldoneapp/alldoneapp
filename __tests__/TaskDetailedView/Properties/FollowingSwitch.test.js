import React from 'react'
import FollowingSwitch from '../../../components/TaskDetailedView/Properties/FollowingSwitch'

import renderer from 'react-test-renderer'

describe('FollowingSwitch component', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<FollowingSwitch />).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it.each([true, false])
        ('should render correctly after call onPress with state "yes" as %p',
            (yesArg) => {
                const tree = renderer.create(<FollowingSwitch />)
                const instance = tree.getInstance()
                instance.state.yes = yesArg

                instance.onPress()
                expect(tree).toMatchSnapshot()
            })
})
