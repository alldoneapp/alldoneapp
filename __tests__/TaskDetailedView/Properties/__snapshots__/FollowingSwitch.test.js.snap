// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowingSwitch component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(78, 93, 120, 1)",
          "fontFamily": "Roboto-Regular",
          "fontSize": 16,
          "letterSpacing": 0.32,
          "lineHeight": 24,
        }
      }
    >
      No
    </Text>
  </View>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(231, 236, 239, 1)",
            "borderColor": "rgba(231, 236, 239, 1)",
            "borderRadius": 21,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 24,
            "paddingLeft": 1,
            "paddingRight": 1,
            "width": 42,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 100,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`FollowingSwitch component should render correctly after call onPress with state "yes" as false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(78, 93, 120, 1)",
          "fontFamily": "Roboto-Medium",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 24,
        }
      }
    >
      Yes
    </Text>
  </View>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(231, 236, 239, 1)",
            "borderColor": "rgba(231, 236, 239, 1)",
            "borderRadius": 21,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 24,
            "paddingLeft": 1,
            "paddingRight": 1,
            "width": 42,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 100,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;

exports[`FollowingSwitch component should render correctly after call onPress with state "yes" as true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(78, 93, 120, 1)",
          "fontFamily": "Roboto-Regular",
          "fontSize": 16,
          "letterSpacing": 0.32,
          "lineHeight": 24,
        }
      }
    >
      No
    </Text>
  </View>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(231, 236, 239, 1)",
            "borderColor": "rgba(231, 236, 239, 1)",
            "borderRadius": 21,
            "borderWidth": 1,
            "flexDirection": "row",
            "height": 24,
            "paddingLeft": 1,
            "paddingRight": 1,
            "width": 42,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 100,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;
